src/ng2-smart-table/components/filter/filter-types/checkbox-filter.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { DefaultFilter } from './default-filter';
+import { Column } from '../../../lib/data-set/column';
+
+@Component({
+  selector: 'checkbox-filter',
+  styles: [`
+    a {
+      font-weight: normal;
+    }
+  `],
+  template: `
+    <input type="checkbox" #filterEl [ngClass]="inputClass" class="form-control">
+    <a href="#" *ngIf="filterActive" (click)="resetFilter($event)">{{column.getFilterConfig()?.resetText || 'reset'}}</a>
+  `
+})
+export class CheckboxFilterComponent extends DefaultFilter implements OnInit {
+
+  filterActive: boolean = false;
+
+  constructor() {
+    super();
+  }
+
+  ngOnInit() {
+    this.changesSubscription = Observable.fromEvent(this.filterEl.nativeElement, 'change')
```

Comment:
We are not against Observables for sure! :) It's just might be not a good practice to use native JS API events when angular provides some wrappers around it. Moreover, I fell like we can combine both observables and angular way, let me get back to you with a link in a moment.